### DISCUSSION TITLE
Is there a better way to list cask which auto-update?

### DISCUSSION THREAD

**Discussion by IdyllicHappiness** [Homebrew/discussions#4033] [Casks] [answered]
I'm quite possibly doing this in an overly convoluted way, but I couldn't find anything in `brew list --help` which would allow me to filter on cask details.

I'd essentially like to list brew casks which are set to auto-update. I run this as part of a helper script which updates various things, I like this output as a little reminder of the auto-updating packages.

It works, but I'd prefer not to have to hit an API multiple times if there's a better solution.

    #!/usr/bin/env bash
    
    # list of casks package names, one per line
    installedCasks=$(brew list -1 --cask)
    
    # get tbe JSON content for each cask from API
    caskJsons=$((xargs -I {} curl -s https://formulae.brew.sh/api/cask/{}.json) <<< "$installedCasks")
    
    # filter JSON to just include name and auto update information
    autoUpdateInformation=$((jq '.token, .auto_updates') <<< "$caskJsons")
    
    # filter on "true", including the line before (the package name)
    autoUpdateCasks=$((rg -B 1 "^true$") <<< "$autoUpdateInformation")
    
    # get rid of any lines which aren't the package names, and remove the surrounding double quotes from package names
    autoUpdateCasksFiltered=$((rg '^"(.*)"$' -r '$1') <<< "$autoUpdateCasks")
    
    echo "$autoUpdateCasksFiltered"

**Comment by p-linnane** [comment]
`grep -r "auto_updates true" $(brew --prefix)/Library/Taps/homebrew/homebrew-cask/Casks`?

**Reply by IdyllicHappiness** [reply]
That's a good shout! It does seem to return all casks in the repo rather than just installed ones, so I went with @gromgit's solution, but this could be useful in certain situations 🙂

**Reply by Qhilm** [reply]
I was looking for a solution without the `jq` dependency and it appears the path for the above command has changed, it does not work for me (macOS 14 & homebrew 4.2.4). I get:

>grep: /opt/homebrew/Library/Taps/homebrew/homebrew-cask/Casks: No such file or directory

I found the below alternative solution (but it is not offline):

`brew list --cask | xargs -I{} brew info homebrew/cask/{} | grep auto_updates`

**Comment by gromgit** [comment, marked as answer]
Since you already have `jq` installed, here's a one-liner that's purely local:
```
brew info --cask --json=v2 $(brew ls --cask) | jq -r '.casks[]|select(.auto_updates==true)|.token'
```

**Reply by IdyllicHappiness** [reply]
Ideal, thank you! (your answer also taught me how to be more succinct with `jq` rather than my hacky grep for `true` with context, followed  by another grep)

**Comment by edbrannin** [comment]
For people like me who came here just wondering what else to `brew upgrade` but forgot to try `brew outdated --help` before googling:

```bash
brew outdated -g
```

**Reply by gromgit** [reply]
Or you could just say "heck, let's just upgrade everything, include the auto-update stuff":
```
brew upgrade --greedy
```
